schematic/schemas/create_json_schema.py

```diff
@@ -0,0 +1,895 @@
+) -> None:
+    """
+    Processes a node in the data model graph.
+    If it should be a property in the JSON Schema, that is set.
+    If it is a property with reverse dependencies, conditional properties are set.
+
+    Argument:
+        json_schema: The JSON Scheme where the node might be set as a property
+        graph_state: The instance tracking the current state of the graph
+        use_property_display_names: If True, the properties in the JSONSchema
+          will be written using node display names
+        use_valid_value_display_names: If True, the valid_values in the JSONSchema
+          will be written using node display names
+    """
+    if graph_state.current_node is None:
+        raise ValueError("Node Processor contains no node.")
+
+    if graph_state.is_current_node_a_property():
+        # Determine if current node has conditional dependencies that need to be set
+        if graph_state.is_current_node_in_reverse_dependencies():
+            _set_conditional_dependencies(
+                json_schema=json_schema,
+                graph_state=graph_state,
+                use_property_display_names=use_property_display_names,
+            )
+            # This is to ensure that all properties that are conditional dependencies are not
+            #   required, but only become required when the conditional dependency is met.
+            graph_state.current_node.is_required = False
+        _set_property(
+            json_schema=json_schema,
+            node=graph_state.current_node,
+            use_property_display_names=use_property_display_names,
+            use_valid_value_display_names=use_valid_value_display_names,
+        )
+        graph_state.update_processed_nodes_with_current_node()
+
+
+def _write_data_model(
+    json_schema_dict: dict[str, Any],
+    schema_path: Optional[str] = None,
+    name: Optional[str] = None,
+    jsonld_path: Optional[str] = None,
+) -> None:
+    """
+    Creates the JSON Schema file
+
+    Arguments:
+        json_schema_dict: The JSON schema in dict form
+        schema_path: Where to save the JSON Schema file
+        jsonld_path:
+          The path to the JSONLD model, used to create the path
+          Used if schema_path is None
+        name:
+          The name of the datatype(source node) the schema is being created for
+          Used if schema_path is None
+    """
+    if schema_path:
+        json_schema_path = schema_path
+    elif name and jsonld_path:
+        json_schema_path = get_json_schema_log_file_path(
+            data_model_path=jsonld_path, source_node=name
+        )
+        json_schema_dirname = os.path.dirname(json_schema_path)
+        if json_schema_dirname != "":
+            os.makedirs(json_schema_dirname, exist_ok=True)
+
+        logger.info(
+            "The JSON schema file can be inspected by setting the following "
+            "nested key in the configuration: (model > location)."
+        )
```

Comment:
I am a bit confused about this log message. If I am reading this code correctly, the JSON Schema file is created in the same directory as the `json_schema_path`, but the message seems to imply that the file won't be available unless you set the location?